llvm/lib/Target/AMDGPU/AMDGPULowerModuleLDSPass.cpp

```diff
@@ -648,7 +649,19 @@ class AMDGPULowerModuleLDS {
         } else if (K.second.size() == 1) {
           KernelAccessVariables.insert(GV);
         } else if (set_is_subset(K.second, HybridModuleRootKernels)) {
-          ModuleScopeVariables.insert(GV);
+          uint64_t LocalMemLimit = 0;
```

Comment:
The comment at the top of the file claims the hybrid strategy offers precise allocation, so is there just a bug somewhere?